Lib/asyncio/tasks.py

```diff
@@ -42,7 +42,9 @@ def all_tasks(loop=None):
     """Return a set of all tasks for the loop."""
     if loop is None:
         loop = events.get_running_loop()
-    return {t for t in _all_tasks
+    # NB: set(_all_tasks) is required to protect
+    # from https://bugs.python.org/issue34970 bug
+    return {t for t in set(_all_tasks)
```

Comment:
Can you make it a `list()` call instead of `set()`?

Reply:
Done